src/CachedCalls.jl

```diff
@@ -39,12 +39,12 @@ macro cached_call(ex)
         ])
         fname = $(cachedcalls_dir()) / "$(h).jlso"
         if isfile(fname)
-            return JLSO.load(fname)[:res]
+            res = JLSO.load(fname)[:res]
         else
             res = $(esc(ex))
             JLSO.save(fname, :res => res)
-            return res
         end
+        res
```

Comment:
no need to do `return res`?

Reply:
I thought so too but it turns out that that's not the case, see https://invenia.slack.com/archives/C0259AJLE/p1614348550040800

The reason is that macros return an expression. When we write
```
variable = @cached_call f()
```
the macro expands to
```
julia> Base.remove_linenums!(@macroexpand @cached_call f())
quote
    var"#410#h" = CachedCalls.hash([f, Any[]...])
    var"#411#fname" = p"/Users/mzgubic/.julia/.cachedcalls" / "$(var"#410#h").jlso"
    if CachedCalls.isfile(var"#411#fname")
        var"#412#res" = ((CachedCalls.JLSO).load(var"#411#fname"))[:res]
    else
        var"#412#res" = f()
        (CachedCalls.JLSO).save(var"#411#fname", :res => var"#412#res")
    end
    var"#412#res"
end
```
which is, eventually:
```
variable = var"#412#res"
```
rather than
```
variable = return var"#412#res"
```